src/v/cluster/members_backend.cc

```diff
@@ -346,11 +382,28 @@ void members_backend::calculate_reallocations_after_node_added(
                   && to_move.left_to_move > 0) {
                     partition_reallocation reallocation(
                       model::ntp(tp_ns.ns, tp_ns.tp, p.id), p.replicas.size());
-
                     reallocation.replicas_to_remove.emplace(to_move.id);
+                    auto current_assignment = p;
```

Comment:
Why do we need this copy? Looks like `reassign_replicas()` first arg can be const ref?

Reply:
`reassign_replicas` mutate the current assignment

Reply:
Ah i'm not sure why I didn't see that before. Makes sense